cmd/atlas/internal/url/url.go

```diff
@@ -0,0 +1,85 @@
+package url
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"ariga.io/atlas/sql/schema"
+)
+
+type (
+	providerMap struct {
+		m map[string]func(string) (*Atlas, error)
+	}
+
+	// Mux is used for routing dsn to correct provider.
+	Mux struct {
+		providers *providerMap
+	}
```

Comment:
```suggestion
	// Mux is used for routing dsn to correct provider.
	Mux struct {
		providers map[string]func(string) (*Atlas, error)
	}
```

Why the `providerMap` is needed?